src/OidcClientSettings.ts

```diff
@@ -92,6 +92,11 @@ export interface OidcClientSettings {
      * Otherwise, they are added to an array as distinct objects for the claim type. (default: false)
      */
     mergeClaims?: boolean;
+    /**
+     * Indicates if the library should use the legacy merge behavior, which mutates string claims into arrays whenever the data is updated on the remote.
+     * This behavior is enabled by default on v2 and will be removed on v3, since it's not a deterministic way of handling claims.
```

Comment:
Please remove "This behavior is enabled by default on v2 and will be removed on v3, since it's not a deterministic way of handling claims."

Maybe we only change the default from `true` to `false` and deprecate it, lets see later...

Reply:
I did already remove this, let me know if you like to add some different message :)

Reply:
For me the legacy behavior looks like a bug!

This merge request fixes that bug. For users who would require the old behavior (i doubt it for now) we provide this option to switch back. In that case the default is `false` and we can already mark it as deprecated.
Like this, we can also will find out sooner if somebody actually requires that strange behavior and for what reason...

I can reflect this by a minor version bump and a special note in the release notes.